src/app/src/test/kotlin/org/imperial/mrc/hint/integration/adr/ADRTests.kt

```diff
@@ -295,6 +295,38 @@ class ADRTests : SecureIntegrationTests()
         }
     }
 
+    @ParameterizedTest
+    @EnumSource(IsAuthorized::class)
+    fun `can overwrite an ADR release of the same name`(isAuthorized: IsAuthorized)
+    {
+        testRestTemplate.postForEntity<String>("/adr/key", getPostEntityWithKey())
+
+        val releaseName = "1.0"
+
+        val result = testRestTemplate.postForEntity<String>(
+                "/adr/datasets/hint_test/releases",
+                getPostEntityWithKey(mapOf("name" to listOf(releaseName)))
+        )
+
+        if (isAuthorized == IsAuthorized.TRUE)
+        {
+            val data = ObjectMapper().readTree(result.body!!)["data"]
+            assertThat(data["name"].textValue()).isEqualTo(releaseName)
+
+            val result2 = testRestTemplate.postForEntity<String>(
+                "/adr/datasets/hint_test/releases",
+                getPostEntityWithKey(mapOf("name" to listOf(releaseName)))
+            )
+
+            val data2 = ObjectMapper().readTree(result2.body!!)["data"]
+            assertThat(data2["name"].textValue()).isEqualTo(releaseName)
```

Comment:
I think worth adding one extra line here, just to check that a new release has actually been created. Otherwise you might just be getting the old one of the same name:
```suggestion
            assertThat(data2["name"].textValue()).isEqualTo(releaseName)
            assertThat(data2["id"].textValue()).isNotEqualTo(data["id"].textValue())
```